test/parallel/test-repl-persistent-history.js

```diff
@@ -258,9 +262,25 @@ function runTest(assertCleaned) {
   const expected = opts.expected;
   const clean = opts.clean;
   const before = opts.before;
+  const deprecated = !!opts.deprecated;
+
+  const depMsg = /NODE_REPL_HISTORY_FILE is deprecated\. Use NODE_REPL_HISTORY instead\./;
 
   if (before) before();
 
+  // TODO: Fix this. There is a bug where env variables/state are shared
+  // between tests. That is why the `if deprecated` is needed, or else tests
+  // will be failing all over the place for no reason. It would better to not
+  // have the guard, because otherwise what if tests that don't have deprecated
+  // set being to have deprecation warnings printed? :(
+  if (deprecated) {
+    common.hijackStderr(function(data) {
+      if (data) {
+        assert.ok(deprecated && depMsg.test(data));
+      }
+    });
+  }
```

Comment:
To me it looks like the hijack is called multiple times and will therefore also trigger for tests that have nothing to do with the current test.
So I think the hijack should be deactivated as soon as it is not required anymore.

In that case the inner `deprecated` check should also be obsolete.